go-kit/metrics/provider/librato/librato.go

```diff
@@ -285,19 +322,40 @@ func (p *Provider) report(u *url.URL, interval time.Duration) error {
 	if err := e.Encode(r); err != nil {
 		return err
 	}
-	resp, err := http.Post(u.String(), "application/json", &buf)
+
+	var rawRequest []byte
+	if p.requestDebugging {
+		rawRequest = buf.Bytes()
```

Comment:
You need to make a copy of the returned slice here as per the docs for Bytes() because http.DefaultClient.Do() will Read() from the buffer.